nuxeo-opencmis-tests/src/test/java/org/nuxeo/ecm/core/opencmis/impl/TestCmisBinding.java

```diff
@@ -2264,6 +2265,65 @@ public void testQueryContains() throws Exception {
+     * specification 1.1 where at most one {@code CONTAINS()} function MUST be included in a single query statement
+     * (section 2.1.14.2.4.4). This relax mode only work for CMISQL query having no {@code JOIN} predicate.
+     *
+     * @see https://jira.nuxeo.com/browse/NXP-19858
+     */
+    @Test
+    @LocalDeploy("org.nuxeo.ecm.core.opencmis.tests.tests:OSGI-INF/test-relax-cmis-spec.xml")
+    public void testQueryMultiContainsRelaxingSpec() throws Exception {
+
+        // when using JOINs we use the CMISQLQueryMaker which hasn't been updated to allow two CONTAINs
+        assumeFalse("JOINs are not supported", supportsJoins());
+
+        ConfigurationService configService = Framework.getService(ConfigurationService.class);
+        assertTrue(configService.isBooleanPropertyTrue(NuxeoRepository.RELAX_CMIS_SPEC));
+
+        ObjectData ob = getObjectByPath("/testfolder1/testfile1");
+        assertEquals("testfile1_Title", getString(ob, "dc:title"));
+
+        PropertyData<?> propTitle = factory.createPropertyStringData("dc:title", "new title1");
+        PropertyData<?> propDescription = factory.createPropertyStringData("dc:description", "new description1");
+        Properties properties = factory.createPropertiesData(Arrays.asList(propTitle, propDescription));
+        Holder<String> objectIdHolder = new Holder<>(ob.getId());
+        Holder<String> changeTokenHolder = getChangeTokenHolder(ob);
+        objService.updateProperties(repositoryId, objectIdHolder, changeTokenHolder, properties, null);
+
+        sleepForFulltext();
+        waitForIndexing();
+        ObjectList res;
+
+        res = query("SELECT cmis:name FROM File WHERE CONTAINS('title1') OR CONTAINS('anotherTitle')");
+        assertEquals(1, res.getNumItems().intValue());
+        assertEquals("new title1", getString(res.getObjects().get(0), PropertyIds.NAME));
+
+        res = query("SELECT cmis:name FROM File WHERE CONTAINS('description1') OR CONTAINS('anotherDescription')");
+        assertEquals(1, res.getNumItems().intValue());
+        assertEquals("new title1", getString(res.getObjects().get(0), PropertyIds.NAME));
+    }
+
+    @Test
+    public void testQueryMultiConainsFollowingSpec() throws Exception {
+
+        ConfigurationService configService = Framework.getService(ConfigurationService.class);
+        assertFalse(configService.isBooleanPropertyTrue(NuxeoRepository.RELAX_CMIS_SPEC));
+
+        ObjectData ob = getObjectByPath("/testfolder1/testfile1");
+        assertEquals("testfile1_Title", getString(ob, "dc:title"));
+
+        String statement = "SELECT cmis:name FROM File WHERE CONTAINS('testfile1_Title') OR CONTAINS('anotherTitle')";
+        try {
+            query(statement);
+            fail();
+        } catch (Exception e) {
+            assertEquals(CmisInvalidArgumentException.class, e.getClass());
+            assertTrue(e.getMessage().contains("At most one CONTAINS() is allowed"));
```

Comment:
`assertTrue(e.getMessage(), e.getMessage().contains("At most one CONTAINS() is allowed"))` so that if it fails you get a meaningful error message.